string new_mode
---
bool success